Surface_mesh/include/CGAL/Surface_mesh/Surface_mesh.h

```diff
@@ -2634,6 +2633,30 @@ namespace boost {
     }
   };
 
+  template <>
+  struct hash<CGAL::SM_Halfedge_index > {
+    std::size_t operator()(const CGAL::SM_Halfedge_index& i) const
+    {
+      return i;
+    }
+  };
+
+  template <>
+  struct hash<CGAL::SM_Edge_index > {
+    std::size_t operator()(const CGAL::SM_Edge_index& i) const
+    {
+      return i;
+    }
+  };
+
+  template <>
+  struct hash<CGAL::SM_Face_index > {
+    std::size_t operator()(const CGAL::SM_Face_index& i) const
+    {
+      return i;
+    }
+  };
+
```

Comment:
I'm surprised it's not needed, I didn't add it just for fun. I think it was to be able to use all the Surface_mesh's descriptors in unordered containers, but it seems it's not needed (anymore (?))... Reverted the commit.